src/main/java/io/supertokens/inmemorydb/queries/SAMLQueries.java

```diff
@@ -0,0 +1,448 @@
+    public static String getQueryToCreateSAMLClientsTable(Start start) {
+        String table = Config.getConfig(start).getSAMLClientsTable();
+        String tenantsTable = Config.getConfig(start).getTenantsTable();
+        // @formatter:off
+        return "CREATE TABLE IF NOT EXISTS " + table + " ("
+                + "app_id VARCHAR(64) NOT NULL DEFAULT 'public',"
+                + "tenant_id VARCHAR(64) NOT NULL DEFAULT 'public',"
+                + "client_id VARCHAR(255) NOT NULL,"
+                + "client_secret TEXT,"
+                + "sso_login_url TEXT NOT NULL,"
+                + "redirect_uris TEXT NOT NULL," // store JsonArray.toString()
+                + "default_redirect_uri VARCHAR(1024) NOT NULL,"
+                + "idp_entity_id VARCHAR(1024),"
+                + "idp_signing_certificate TEXT,"
+                + "allow_idp_initiated_login BOOLEAN NOT NULL DEFAULT FALSE,"
+                + "enable_request_signing BOOLEAN NOT NULL DEFAULT TRUE,"
+                + "created_at BIGINT NOT NULL,"
+                + "updated_at BIGINT NOT NULL,"
+                + "UNIQUE (app_id, tenant_id, idp_entity_id),"
+                + "PRIMARY KEY (app_id, tenant_id, client_id),"
+                + "FOREIGN KEY (app_id, tenant_id) REFERENCES " + tenantsTable + " (app_id, tenant_id) ON DELETE CASCADE"
+                + ");";
+        // @formatter:on
+    }
+
+    public static String getQueryToCreateSAMLClientsAppIdTenantIdIndex(Start start) {
+        String table = Config.getConfig(start).getSAMLClientsTable();
+        return "CREATE INDEX IF NOT EXISTS saml_clients_app_tenant_index ON " + table + "(app_id, tenant_id);";
+    }
+
+    public static String getQueryToCreateSAMLRelayStateTable(Start start) {
+        String table = Config.getConfig(start).getSAMLRelayStateTable();
+        String tenantsTable = Config.getConfig(start).getTenantsTable();
+        // @formatter:off
+        return "CREATE TABLE IF NOT EXISTS " + table + " ("
+                + "app_id VARCHAR(64) NOT NULL DEFAULT 'public',"
+                + "tenant_id VARCHAR(64) NOT NULL DEFAULT 'public',"
+                + "relay_state VARCHAR(255) NOT NULL,"
+                + "client_id VARCHAR(255) NOT NULL,"
+                + "state TEXT," // nullable
+                + "redirect_uri VARCHAR(1024) NOT NULL,"
+                + "created_at BIGINT NOT NULL,"
+                + "expires_at BIGINT NOT NULL,"
+                + "PRIMARY KEY (relay_state)," // relayState must be unique
+                + "FOREIGN KEY (app_id, tenant_id) REFERENCES " + tenantsTable + " (app_id, tenant_id) ON DELETE CASCADE"
+                + ");";
+        // @formatter:on
+    }
+
+    public static String getQueryToCreateSAMLRelayStateAppIdTenantIdIndex(Start start) {
+        String table = Config.getConfig(start).getSAMLRelayStateTable();
+        return "CREATE INDEX IF NOT EXISTS saml_relay_state_app_tenant_index ON " + table + "(app_id, tenant_id);";
+    }
```

Comment:
We expect this table to grow, and it could grow rather large. Don't we need an index on `expires_at` for cleanup?